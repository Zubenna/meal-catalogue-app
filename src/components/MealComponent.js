import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Style from '../styles/MealComponentStyle.module.css';

const MealComponent = (props) => {
  const { meal } = props;
  const uniqueKey = meal.idMeal;
  return (
    <div>
      <div className={Style.mealDetails} key={uniqueKey}>
        <div className={Style.imgContainer}>
          <img src={meal.strMealThumb} data-testid="image" alt={meal.strMeal} className={Style.img} />
        </div>
        <div className={Style.mealName}>
          <h3 data-testid="name">{meal.strMeal}</h3>
        </div>
        <button type="button">
          <Link className={Style.link} to={`/meal/${uniqueKey}`}>
            <h4 className={Style.detail}>See Details</h4>
          </Link>
        </button>
      </div>
    </div>
  );
};

MealComponent.propTypes = {
  meal: PropTypes.shape({
    strMealThumb: PropTypes.string,
    strMeal: PropTypes.string,
    idMeal: PropTypes.string,
  }).isRequired,
};

export default MealComponent;
