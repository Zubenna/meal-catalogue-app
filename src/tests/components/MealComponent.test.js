import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import MealComponent from '../../components/MealComponent';

describe('<MealComponent />', () => {
  const  meal = {
    meal: [{ name: 'Christmas Pudding Flapjack', area: 'Britain' }, { name: 'Chicken Handi', area: 'Indian' }],
  };

  it('it renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <MealComponent meal={meal} />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an image', () => {
    const { queryAllByTestId } = render(
      <Router>
        <MealComponent meal={meal} />
      </Router>,
    );
    const image = queryAllByTestId('image');
    expect(image[0]).toBeTruthy();
  });

  it('renders a name', () => {
    const { queryAllByTestId } = render(
      <Router>
       <MealComponent meal={meal} />
      </Router>,
    );
    const name = queryAllByTestId('name');
    expect(name[0]).toBeTruthy();
  });

  it('renders an image corecctly', () => {
    const { queryAllByTestId } = render(
      <Router>
       <MealComponent meal={meal} />
      </Router>,
    );
    const image = queryAllByTestId('img');
    expect(image[0]).toBeFalsy();
  });

  it('renders a name', () => {
    const { queryAllByTestId } = render(
      <Router>
        <MealComponent meal={meal} />
      </Router>,
    );
    const name = queryAllByTestId('name');
    expect(name[0]).not.toBeFalsy();
  });
});