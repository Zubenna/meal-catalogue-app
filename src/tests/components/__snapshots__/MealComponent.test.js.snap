// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealComponent /> it renders correctly 1`] = `
<tbody>
  <tr>
    <td>
      <div
        className="mealDetails"
      >
        <div
          className="imgContainer"
        >
          <img
            className="img"
            data-testid="image"
          />
        </div>
        <div
          className="mealName"
        >
          <h3
            data-testid="name"
          />
        </div>
        <button
          type="button"
        >
          <a
            className="link"
            href="/meal/undefined"
            onClick={[Function]}
          >
            <h4
              className="detail"
            >
              See Details
            </h4>
          </a>
        </button>
      </div>
    </td>
  </tr>
</tbody>
`;
